### PR TITLE
Address feedback from previous PRs

### PR DESCRIPTION
**Pull request by adrw**
Address comments from PR: https://github.com/cashapp/barber/pull/52

### CODE REVIEW

File: barber/src/main/kotlin/app/cash/barber/Barber.kt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package app.cash.barber
 
 import app.cash.barber.models.Document
-import app.cash.barber.models.Locale
+import app.cash.barber.locale.Locale
```
Comment (adrw) [inline barber/src/main/kotlin/app/cash/barber/Barber.kt:4]: I moved all `Locale` related classes into a new `Locale` package for better code organization.

File: barber/src/main/kotlin/app/cash/barber/models/MapToMap.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package app.cash.barber.models
+
+fun <T, K, V> Iterable<T>.mapToMap(entryFactory: (T) -> Pair<K, V>): Map<K, V> {
```
Comment (ejao1) [inline barber/src/main/kotlin/app/cash/barber/models/MapToMap.kt:3]: Can this be replaced by https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.sequences/associate.html ?

File: barber/src/main/kotlin/app/cash/barber/BarbershopBuilder.kt [hunk 3]
```diff
@@ -205,15 +207,16 @@ class BarbershopBuilder : Barbershop.Builder {
...
-          mapOf<RealBarber.DocumentTemplateKey, DocumentTemplateDb>()) { acc, (key, db) ->
-        val (_, locale, sourceBarberSignature, version) = key
-        acc + mapOf(RealBarber.DocumentTemplateKey(
-            locale = locale,
-            sourceBarberSignature = sourceBarberSignature,
-            version = version
-        ) to db)
-      }
+      val localeVersionsMap: Map<RealBarber.DocumentTemplateKey, DocumentTemplateDb> =
+          versions.entries.mapToMap { (key, db) ->
```
Comment (swankjesse) [inline barber/src/main/kotlin/app/cash/barber/BarbershopBuilder.kt:211]: nice name

File: barber/src/test/kotlin/app/cash/barber/locale/LocaleResolverTest.kt [hunk 4]
```diff
@@ -85,17 +90,46 @@ class LocaleResolverTest {
...
+        |LocaleResolver: class app.cash.barber.locale.BadMapleSyrupLocaleResolver
         |Locale: [Locale=en-CA]
         |Resolved Locale: [Locale=en-CA]
         |
       """.trimMargin(),
         exception.toString())
   }
+
+  @Test
+  fun `LocaleResolver works with a Guice Table`() {
```
Comment (swankjesse) [inline barber/src/test/kotlin/app/cash/barber/locale/LocaleResolverTest.kt:108]: Guava table?

File: barber/src/test/kotlin/app/cash/barber/locale/LocaleResolverTest.kt [hunk 5]
```diff
@@ -85,17 +90,46 @@ class LocaleResolverTest {
...
+    table.put(EN_CA, 2, DocumentTemplate(template_token = "t4"))
+    val expectedMap = mapOf(
+        1L to DocumentTemplate(template_token = "t3"),
+        2L to DocumentTemplate(template_token = "t4")
+    )
+    assertThat(resolver.resolve(EN_US, table)).containsAllEntriesOf(expectedMap)
+  }
+
+  @Test
+  fun `LocaleResolver fails with empty Guice Table`() {
```
Comment (swankjesse) [inline barber/src/test/kotlin/app/cash/barber/locale/LocaleResolverTest.kt:123]: Guava table?

**Review by swankjesse — APPROVED** [review]

File: barber/src/test/kotlin/app/cash/barber/locale/LocaleResolverTest.kt (hunk 5 above)
Comment (adrw) [reply barber/src/test/kotlin/app/cash/barber/locale/LocaleResolverTest.kt:123]: Whoops, thanks!

File: barber/src/main/kotlin/app/cash/barber/BarbershopBuilder.kt (hunk 3 above)
Comment (adrw) [reply barber/src/main/kotlin/app/cash/barber/BarbershopBuilder.kt:211]: Thanks partner 🤠 !

File: barber/src/test/kotlin/app/cash/barber/locale/LocaleResolverTest.kt [hunk 6]
```diff
@@ -85,17 +90,46 @@ class LocaleResolverTest {
...
+    val resolver = MapleSyrupOrFirstLocaleResolver()
+    val table = HashBasedTable.create<Locale, Long, DocumentTemplate>()
+    val exception = assertFailsWith<BarberException> {
+      resolver.resolve(EN_US, table)
+    }
+    assertEquals(exception.toString(), """
+      |Errors
```
[the remark this answers is not in the document]
Comment (adrw) [reply barber/src/test/kotlin/app/cash/barber/locale/LocaleResolverTest.kt:130]: It has helped ensure backwards compatibility with previous releases though it definitely is at times annoyingly brittle.

File: barber/src/test/kotlin/app/cash/barber/locale/BadMapleSyrupLocaleResolver.kt [hunk 7]
```diff
@@ -0,0 +1,9 @@
+package app.cash.barber.locale
+
+import app.cash.barber.locale.Locale.Companion.EN_CA
+
+class BadMapleSyrupLocaleResolver : LocaleResolver {
+  override fun resolve(locale: Locale, options: Set<Locale>): Locale {
+    return EN_CA
```
[the remark this answers is not in the document]
Comment (adrw) [reply barber/src/test/kotlin/app/cash/barber/locale/BadMapleSyrupLocaleResolver.kt:7]: 🇨🇦

**Pull request merged**